File: src/test/java/com/morak/performancetracker/aop/persistence/QueryMonitorTest.java
package com.morak.performancetracker.aop.persistence;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatThrownBy;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

class QueryMonitorTest {

    @Test
    void 쿼리시간_측정을_시작한다() {
        // given
        QueryMonitor monitor = new QueryMonitor();
        // when
        monitor.start();
        // then
        assertThat(monitor.getStartTime()).isNotZero();
    }

    @Test
    void 쿼리시간_측정을_종료한다() {
        // given
        QueryMonitor monitor = new QueryMonitor();
        monitor.start();
        // when
        monitor.end();
        // then
        assertThat(monitor.getQueryTime()).isNotZero();
    }

    @Test
    void 쿼리시간_측정종료시_시작시간이_0이상이어야_한다() {
        // given
        QueryMonitor monitor = new QueryMonitor();
        // when & then
        assertThatThrownBy(() -> monitor.end()).isInstanceOf(IllegalStateException.class);
    }

    @Test
    void 모니터를_초기화한다() {
        // given
        QueryMonitor monitor = new QueryMonitor();
        // when
        monitor.clear();
        // then
        Assertions.assertAll(
                () -> assertThat(monitor.getQueryTime()).isZero(),
                () -> assertThat(monitor.getQueryCount()).isZero(),
                () -> assertThat(monitor.getStartTime()).isZero()
        );
    }
}

File: src/test/kotlin/com/morak/performancetracker/aop/persistence/QueryMonitorTest.kt
package com.morak.performancetracker.aop.persistence

import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test

class QueryMonitorTest {
    @Test
    fun `쿼리시간 측정을 시작한다`() {
        // given
        val monitor = QueryMonitor()
        // when
        monitor.start()
        // then
        Assertions.assertThat(monitor.startTime).isNotZero()
    }

    @Test
    fun `쿼리시간 측정을 종료한다`() {
        // given
        val monitor = QueryMonitor()
        monitor.start()
        // when
        monitor.end()
        // then
        Assertions.assertThat(monitor.queryTime).isNotZero()
    }

    @Test
    fun `쿼리시간 측정종료시 시작시간이 0이상이어야 한다`() {
        // given
        val monitor = QueryMonitor()
        // when & then
        Assertions.assertThatThrownBy { monitor.end() }.isInstanceOf(IllegalStateException::class.java)
    }

    @Test
    fun `모니터를 초기화한다`() {
        // given
        val monitor = QueryMonitor()
        // when
        monitor.clear()
        // then
        org.junit.jupiter.api.Assertions.assertAll(
            { Assertions.assertThat(monitor.queryTime).isZero() },
            { Assertions.assertThat(monitor.startTime).isZero() }
        )
    }
}
